examples/02_first_level_models/plot_localizer_surface_analysis.py

```diff
@@ -1,117 +1,190 @@
 """
-First level analysis of localizer dataset
-=========================================
+Example of surface-based first-level analysis
+=============================================
 
-Full step-by-step example of fitting a GLM to experimental data and visualizing
-the results.
+Full step-by-step example of fitting a GLM to experimental data sampled on the cortical surface and visualizing the results.
 
 More specifically:
 
 1. A sequence of fMRI volumes are loaded
-2. A design matrix describing all the effects related to the data is computed
-3. a mask of the useful brain volume is computed
+2. fMRI data are projected onto a reference cortical surface (the freesurfer template, fsaverage)
+3. A design matrix describing all the effects related to the data is computed
 4. A GLM is applied to the dataset (effect/covariance,
    then contrast estimation)
 
+The result of the analysis are statistical maps that are defined on the brain mesh. We disply them using Nilearn capabilities.
+
+The projection of fMRI data onto a given brain mesh requires that both are initially defined in the same space. 
+
+* The functional data should be coregistered to the anatomy from which the mesh was obtained. 
+
+* Another possibility, used here, is to project the normalized fMRI data to an MNI-coregistered mesh, such as fsaverage. 
+
+The advantage of this second approach is that it makes it easy to run second-level analyses on the surface. On the other hand, it is obviously less accurate than using a subject-tailored mesh.
+
+
 """
-import numpy as np
-import pandas as pd
-import nilearn
 
 #########################################################################
 # Prepare data and analysis parameters
 # -------------------------------------
-# Prepare timing
+# Prepare timing parameters
 t_r = 2.4
 slice_time_ref = 0.5
 
+#########################################################################
 # Prepare data
+# First the fMRI data
 from nistats.datasets import fetch_localizer_first_level
 data = fetch_localizer_first_level()
+fmri_img = data.epi_img
+
+#########################################################################
+# Second the experimental paradigm
 paradigm_file = data.paradigm
+import pandas as pd
 paradigm = pd.read_csv(paradigm_file, sep=' ', header=None, index_col=None)
 paradigm.columns = ['session', 'trial_type', 'onset']
-fmri_img = data.epi_img
+
 
 #########################################################################
 # Project the fMRI image to the surface
 # -------------------------------------
+#
+# For this we need to get a mesh representing the geometry of the surface.
+# we could use an individual mesh, but we first resort to a standard mesh, the so-called fsaverage5 template from the Freesurfer software.
 
+import nilearn
 fsaverage = nilearn.datasets.fetch_surf_fsaverage5()
+
```

Comment:
Above: VisibleDeprecationWarning: fetch_surf_fsaverage5 has been deprecated and will be removed in a future release. Use fetch_surf_fsaverage(mesh='fsaverage5')

Reply:
To avoid annoying deprecation warnings.